docs/devguide/messenger_overview.rst

```diff
@@ -0,0 +1,147 @@
+The reply queue is automatically generated and disposed of at the end of the request so no subscriptions need to be managed by the consumer.
+
+Object Marshaling
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+While plain JavaScript objects can be sent over the messenger it also supports marshaling of Serializable types in On-Core. Objects which implement the Serializable interface can be marshaled over AMQP by using a constructor initialization convention and by registering their type with the messenger.
+When sending a Serializable object over AMQP the messenger uses the registered type to decorate the AMQP message in a way in which a receiver can create a new copy of the object using it's typed constructor.
+Subscribers who receive constructed types will have access to them directly through their data value in the subscriber callback.
+
+Object Validation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+On publish and on subscription callback the messenger will also validate Serializable objects using the Validatable base class.
+Validation is provided via JSON Schemas which are attached to the sub-classed Validatable objects.
+If an object to be marshaled is Validatable the messenger will validate the object prior to publish or subscribe callback.
+Future versions of the messenger will support subscription and request type definitions which will allow consumers to identify what types of objects they expect to be notified about which will give the messenger an additional means of ensuring communications are handled correctly.
+Some example schemas are listed below:
+MAC Address
+
+.. code-block:: javascript
+
+    {
+        id: 'MacAddress',
+        type: 'object',
+        properties: {
+            value: {
+                type: 'string',
+                pattern: '^([0-9a-fA-F][0-9a-fA-F]:){5}([0-9a-fA-F][0-9a-fA-F])$'
+            }
+        },
+        required: [ 'value' ]
+    }
+
+IP Address
+
+.. code-block:: javascript
+
+    {
+        id: 'IpAddress',
+        type: 'object',
+        properties: {
+            value: {
+                type: 'string',
+                format: 'ipv4'
+            }
+        },
+        required: [ 'value' ]
+    }
+
+Lookup Model (via On-Http)
+
+.. code-block:: javascript
+
+
+    {
+        id: 'Serializables.V1.Lookup',
+        type: 'object',
+        properties: {
+            node: {
+                type: 'string'
+            },
+            ipAddress: {
+                type: 'string',
+                format: 'ipv4'
+            },
+            macAddress: {
+                type: 'string',
+                pattern: '^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$'
+            }
+        },
+        required: [ 'macAddress' ]
+    }
+
+Tracing
```

Comment:
We gutted this because of the perf impact vs. measurable value, so you should remove the "Tracing" section.